### PR TITLE
Disable tagged names

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26366 Refactor Dimname.h API to be nicer
* #26365 Delete tagged names
* **#26356 Disable tagged names**

This PR doesn't delete the code for them yet because it takes some effort to
determine what to delete. I will send a followup PR fully deleting
tagged names, but this PR disables their creation.

Test Plan:
- [namedtensor ci]

Pull Request resolved: https://github.com/pytorch/pytorch/pull/26356

Differential Revision: [D17428409](https://our.internmc.facebook.com/intern/diff/D17428409)

### CODE REVIEW

File: aten/src/ATen/core/Dimname.cpp [hunk 1]
```diff
@@ -54,26 +54,8 @@ Dimname Dimname::fromSymbol(Symbol full_name) {
...
-  TORCH_CHECK(
-      second_dot == std::string::npos,
-      "Invalid name '", str, "': A tagged name can only contain one '.'");
-  auto untagged_name = str.substr(0, it);
-  auto tag = str.substr(it + 1);
-  check_valid_identifier(untagged_name); 
-  check_valid_identifier(tag);
-  return Dimname(NameType::TAGGED, full_name, Symbol::dimname(untagged_name));
+  check_valid_identifier(full_name.toUnqualString());
+  return Dimname(full_name);
```
Comment (nairbv) [inline aten/src/ATen/core/Dimname.cpp]: if we're getting rid of tagged names, should it just be "name" instead of "full_name"?

Comment (zou3519) [reply aten/src/ATen/core/Dimname.cpp]: I changed that in the next diff in the stack: https://github.com/pytorch/pytorch/pull/26365.

This one is to just disable creation of tagged names.

Comment (zou3519) [reply aten/src/ATen/core/Dimname.cpp]: I can merge the two PRs together (perhaps they make more sense as one unit) if that makes them easier to review.

File: aten/src/ATen/test/Dimname_test.cpp [hunk 2]
```diff
@@ -31,32 +31,17 @@ TEST(DimnameTest, wildcardName) {
...
 }
 
 TEST(DimnameTest, createNormalName) {
   auto foo = Symbol::dimname("foo");
   auto dimname = Dimname::fromSymbol(foo);
   ASSERT_EQ(dimname.type(), NameType::NORMAL);
   ASSERT_EQ(dimname.full_name(), foo);
-  ASSERT_EQ(dimname.untagged_name(), foo);
-
+  ASSERT_THROW(Dimname::fromSymbol(Symbol::dimname("inva.lid")), c10::Error);
```
Comment (nairbv) [inline aten/src/ATen/test/Dimname_test.cpp]: what are we throwing in this case? that `.` isn't allowed?

File: aten/src/ATen/core/Dimname.cpp (hunk 1 above)
Comment (nairbv) [reply aten/src/ATen/core/Dimname.cpp]: yup, saw it.

**Review by nairbv — APPROVED** [review]

File: aten/src/ATen/test/Dimname_test.cpp (hunk 2 above)
Comment (zou3519) [reply aten/src/ATen/test/Dimname_test.cpp]: We are throwing the error message that says "only alphabetical and underscore characters are allowed"

**Pull request closed without merging**